Source/console-checkers/Source/ICommand.h

```diff
@@ -0,0 +1,175 @@
+//---------------------------------------------------------------
+//
+// ICommand.h
+//
+
+#pragma once
+
+#include "GameTypes.h"
+
+#include <string>
+
+namespace Checkers {
+
+//===============================================================
+
+class Game;
+
+struct CommandErrorInfo {
+	// Player-facing reason as to why this command failed to process.
+	std::string errorReason;
+};
+
+class ICommand {
+public:
+	ICommand(const ICommand& other) = delete;
+	ICommand& operator=(const ICommand& other) = delete;
+	ICommand(ICommand&& other) noexcept = default;
+	ICommand& operator=(ICommand&& other) noexcept = default;
+
+	ICommand() = default;
+	virtual ~ICommand();
+
+	// Each command knows how to execute itself.
+	virtual bool Execute(Game* game) = 0;
```

Comment:
I hate SUPER generically named functions like this. Try searching for Execute and you get way too much. ExecuteCommand even if redundant, helps a ton.